apps/dc_tools/odc/apps/dc_tools/export_md.py

```diff
@@ -0,0 +1,264 @@
+"""
+Metadata Transformation from old format to new format:
+Example usage:
+ python metadata_transformer.py --datacube-config $HOME/.datacube.conf transform --product aster_l1t_swir
+    --config metadata_transform_config.yaml --output-dir /g/data/u46/users/aj9439/metadata
+    --limit 10
+The config file contains the grid mappings to band names:
+    grids:
+      default: ['3', '4', '5']
+      swir: ['1', '2']
+"""
+from pathlib import Path
+import yaml
+import click
+
+from datacube.testutils.io import native_geobox
+from datacube import Datacube
+
+
+@click.group(help=__doc__)
+@click.option('--datacube-config', '-c', help="Pass the configuration file to access the database",
+              type=click.Path(exists=True))
+@click.pass_context
+def cli(ctx, datacube_config):
+    """ Used to pass the datacube index to functions via click."""
+    ctx.obj = Datacube(config=datacube_config).index
+
+
+@cli.command()
+@click.option('--product', required=True, help='Which product?')
+@click.option('--config', required=True, help='The configuration file for transformation')
+@click.option('--output-dir', required=True, help='New metadata yaml file is written to this dir')
+@click.option('--limit', help='maximum number of datasets to process')
+@click.pass_obj
+def transform(index, product, config, output_dir, limit):
+
+    # Get the product
+    dataset_type = index.products.get_by_name(product)
+
+    with open(config, 'r') as config_file:
+        cfg = yaml.load(config_file)
+
+    # Is this a ingested product?
+    if dataset_type.grid_spec is not None:
+        transform_ingested_datasets(index, product, cfg, Path(output_dir), limit)
+    else:
+        transform_indexed_datasets(index, product, cfg, Path(output_dir), limit)
+
+
+def transform_ingested_datasets(index, product, config, output_dir, limit):
+    """
+    Transform the metadata of ingested product. The product-wide fixed sections
+    of metadata such as 'grids' is computed just once.
+    """
+
+    dataset_ids = index.datasets.search_returning(limit=limit, field_names=('id',), product=product)
+
+    # Compute 'grids' section
+    dataset_id = next(dataset_ids)
```

Comment:
This code duplication is a bit confusing.

Would be easier to read in this format:

```python
for cc, datasets_id in enumerate(datasets_ids):
    if cc == 0:
      ..do_setup_things..
    ..do_usual_things..
```

This is python `if` inside loop is not as costly in relative terms.

Reply:
Done